clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -825,27 +826,24 @@ public void seek(TopicPartition partition, OffsetAndMetadata offsetAndMetadata)
 
     @Override
     public void seekToBeginning(Collection<TopicPartition> partitions) {
-        if (partitions == null)
-            throw new IllegalArgumentException("Partitions collection cannot be null");
-
-        acquireAndEnsureOpen();
-        try {
-            Collection<TopicPartition> parts = partitions.isEmpty() ? subscriptions.assignedPartitions() : partitions;
-            subscriptions.requestOffsetReset(parts, OffsetResetStrategy.EARLIEST);
-        } finally {
-            release();
-        }
+        seekWithResetEventInternal(partitions, OffsetResetStrategy.EARLIEST);
     }
 
     @Override
     public void seekToEnd(Collection<TopicPartition> partitions) {
+        seekWithResetEventInternal(partitions, OffsetResetStrategy.LATEST);
+    }
+
+    private void seekWithResetEventInternal(Collection<TopicPartition> partitions, OffsetResetStrategy offsetResetStrategy) {
```

Comment:
Suggestion: I think we could name this method simply `seek()` since the parameters are unique.